src/bin/rav1e-ch.rs

```diff
@@ -0,0 +1,574 @@
+// Copyright (c) 2017-2019, The rav1e contributors. All rights reserved
+//
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+#![deny(bare_trait_objects)]
+#![allow(clippy::cast_lossless)]
+#![allow(clippy::cast_ptr_alignment)]
+#![allow(clippy::cognitive_complexity)]
+#![allow(clippy::needless_range_loop)]
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::verbose_bit_mask)]
+#![allow(clippy::unreadable_literal)]
+#![allow(clippy::many_single_char_names)]
+#![warn(clippy::expl_impl_clone_on_copy)]
+#![warn(clippy::linkedlist)]
+#![warn(clippy::map_flatten)]
+#![warn(clippy::mem_forget)]
+#![warn(clippy::mut_mut)]
+#![warn(clippy::mutex_integer)]
+#![warn(clippy::needless_borrow)]
+#![warn(clippy::needless_continue)]
+#![warn(clippy::path_buf_push_overwrite)]
+#![warn(clippy::range_plus_one)]
+
+#[macro_use]
+extern crate log;
+
+mod common;
+mod decoder;
+mod error;
+#[cfg(feature = "serialize")]
+mod kv;
+mod muxer;
+mod stats;
+
+use crate::common::*;
+use crate::error::*;
+use crate::stats::*;
+use rav1e::prelude::*;
+
+use crate::decoder::{Decoder, FrameBuilder, VideoDetails};
+use crate::muxer::*;
+use std::fs::File;
+use std::io::{Read, Seek, Write};
+use std::sync::Arc;
```

Comment:
Also maybe structure the imports into two blocks. One with crate, the other without.